### PR TITLE
workqueue: use typed queue

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
Replace deprecated RateLimitingInterface and similar with the typed workqueue ones.
This allows us to drop some checks and type assertions. Converted the mockqueue using generics.

Fixes #

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by 0xFelix — APPROVED** [review]
From what I can tell:

/lgtm

**Review by iholder101 — COMMENTED** [review]
Thank you @fossedihelm! Great stuff

/approve

File: pkg/storage/export/export/export.go [hunk 1]
```diff
@@ -262,7 +262,10 @@ func (ctrl *VMExportController) Init() error {
 	if err != nil {
 		return err
 	}
-	ctrl.vmExportQueue = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "virt-controller-export-vmexport")
+	ctrl.vmExportQueue = workqueue.NewTypedRateLimitingQueueWithConfig[string](
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/storage/export/export/export.go:265]: Oh wow! Thank you! Will do it in a follow-up :)

**Pull request merged**